pkg/runtime/unstructured/converter.go

```diff
@@ -0,0 +1,472 @@
+/*
```

Comment:
@smarterclayton - I tried a bunch of other optimizations but most of them were useless, so I removed them as they were complicating code. In the current shape we have:

wojtekt@wojtekt-work:~/go/src/k8s.io/kubernetes$ go test ./pkg/runtime/unstructured -test.run=XXXXX -bench=ToFrom -benchmem -memprofile mem.out -memprofilerate=1
BenchmarkToFromUnstructured-12           	    2000	   1318675 ns/op	   15716 B/op	     295 allocs/op
BenchmarkToFromUnstructuredViaJSON-12    	    1000	   2055132 ns/op	   29659 B/op	     457 allocs/op
PASS
ok  	k8s.io/kubernetes/pkg/runtime/unstructured	5.502s
wojtekt@wojtekt-work:~/go/src/k8s.io/kubernetes$ go test ./pkg/runtime/unstructured -test.run=XXXXX -bench=ToFrom
BenchmarkToFromUnstructured-12           	   10000	    122756 ns/op
BenchmarkToFromUnstructuredViaJSON-12    	    5000	    206080 ns/op
PASS
ok  	k8s.io/kubernetes/pkg/runtime/unstructured	2.484s

Which is not ideal, but I think worth merging. We can probably try optimizing further later.

Can you please take a look?

Also, I need some guidance where I should put the code now with all these moves to separate repositories. Should it go to staging/ directory? If so, how/when it will be available for use e.g. by test in pkg/api ?